addons/product/models/pricelist.py

```diff
@@ -0,0 +1,322 @@
+        help='Pricelist Item applicable on selected option', default='3_global')
+    sequence = fields.Integer(required=True, help="Gives the order in which the pricelist items will be checked. The evaluation gives highest priority to lowest sequence and stops as soon as a matching item is found.", default=5)
+    base = fields.Selection([('list_price', 'Public Price'), ('standard_price', 'Cost'), ('pricelist', 'Other Pricelist')], string="Based on", required=True,
+        help='Base price for computation. \n Public Price: The base price will be the Sale/public Price. \n Cost Price : The base price will be the cost price. \n Other Pricelist : Computation of the base price based on another Pricelist.', default='list_price')
+    base_pricelist_id = fields.Many2one('product.pricelist', string='Other Pricelist')
+    pricelist_id = fields.Many2one('product.pricelist', string='Pricelist')
+    price_surcharge = fields.Float(string='Price Surcharge', digits=dp.get_precision('Product Price'),
+        help='Specify the fixed amount to add or substract(if negative) to the amount calculated with the discount.')
+    price_discount = fields.Float(string='Price Discount', digits=(16, 2), default=0.0)
+    price_round = fields.Float(string='Price Rounding', digits=dp.get_precision('Product Price'),
+        help="Sets the price so that it is a multiple of this value.\n" \
+            "Rounding is applied after the discount and before the surcharge.\n" \
+            "To have prices that end in 9.99, set rounding 10, surcharge -0.01")
+    price_min_margin = fields.Float(string='Min. Price Margin', digits=dp.get_precision('Product Price'),
+        help='Specify the minimum amount of margin over the base price.')
+    price_max_margin = fields.Float(string='Max. Price Margin', digits=dp.get_precision('Product Price'),
+        help='Specify the maximum amount of margin over the base price.')
+    company_id = fields.Many2one('res.company', related='pricelist_id.company_id', readonly=True, string='Company', store=True)
+    currency_id = fields.Many2one('res.currency', related='pricelist_id.currency_id', readonly=True, string='Currency', store=True)
+    date_start = fields.Date(string='Start Date', help="Starting date for the pricelist item validation")
+    date_end = fields.Date(string='End Date', help="Ending valid for the pricelist item validation")
+    compute_price = fields.Selection([('fixed', 'Fix Price'), ('percentage', 'Percentage (discount)'), ('formula', 'Formula')], index=True, default='fixed')
+    fixed_price = fields.Float(string='Fixed Price')
+    percent_price = fields.Float(string='Percentage Price')
+
+    @api.constrains('price_min_margin', 'price_max_margin')
+    def _check_margin(self):
+        if self.price_min_margin > self.price_max_margin:
```

Comment:
Use self.ensure_one() to verify that the self contains only one record or make the method to work on multiple records.